packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -67,6 +89,31 @@ Each column that is sorted will have an appropriate icon displayed in the header
 
 Each column that is sorted ascending will have `aria-sort="ascending"` set on the header, and each column that is sorted descending will have `aria-sort="descending"` set on the header.
 
+### Interactive Sorting UX
+
+The UX/ interactions for interactive sorting are captured in [#885: UX Design for basic column/row interactions](https://github.com/ni/nimble/issues/885).  
+To summarize:
+
+If the table `sortMode` is `single` or `multiple`:  
+Single-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were also sorted become unsorted
+-   The clicked column gets a sort index of `0` when sorted, and `undefined` if it's transitioning back to unsorted.
+
+If the table `sortMode` is `multiple` only:  
+Shift-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were already sorted maintain their sorting
+-   If the clicked column was already sorted, its sort index does not change
+-   If the clicked column is transitioning to unsorted, its sort index will be set to `undefined`
+-   If the clicked column is transitioning to being sorted, it will get a sort index of `0` if no other columns were already sorted, or `max(alreadySortedColumn1SortIndex, ..., alreadySortedColumnNSortIndex) + 1` if there were existing sorted columns
+
+If interactive sorting is enabled, sorting menu items also appear in the column header menu:  
+![Sorting via Column Header Menu](./spec-images/HeaderMenuSorting.png)  
+Sorting via the menu will always unsort any other columns that were already sorted (TBD: even when unsorting the current column?)
+
+For columns with `sorting-disabled` set to true, clicking/Shift-clicking the column header will not affect the sort state, and the sort menu options will not appear in the column header menu.
```

Comment:
It's your call how to stage the implementation, but I'd be ok with defering the column header menu to a separate PR after the header click interactions since it's not an SLE parity feature. We could even consider splitting it into a separate feature that's not part of Milestone 4. But I think the detail you have about it here is good to keep.

Reply:
Yeah I think that the column header menu would be a follow up PR (I added a separate task row for it in #854)